modules/apps/static/portal-osgi-web/portal-osgi-web-servlet-jsp-compiler/src/main/java/com/liferay/portal/osgi/web/servlet/jsp/compiler/internal/JSPTaglibHelperImpl.java

```diff
@@ -55,12 +56,13 @@ public void scanTLDs(
 
 		BundleWiring bundleWiring = bundle.adapt(BundleWiring.class);
 
-		Collection<String> resources = bundleWiring.listResources(
-			"META-INF/", "*.tld", BundleWiring.LISTRESOURCES_RECURSE);
+		Collection<String> resources = new ArrayList<>(
```

Comment:
Hi @dantewang @yuhai ,
Result of findEntries() should cover result of findResources(), we should not do same thing twice.

Tina.

Reply:
@tinatian findResources() is backed by classloader. It can find things e.g. in a jar located in WEB-INF/lib of a WAR, while findEntries() can't.

Reply:
@dantewang Aha, I see...

Tina.